### ISSUE TITLE
[Feature Request] Ability to merge two songs that are literally the same recording, probably on different albums

### ISSUE THREAD

**Issue by 2q2code** [Yooooomi/your_spotify#465]
When viewing stats on an artist, the top charts are segmented incorrectly if a song you listen to a lot is contained in different albums somehow (maybe?). So a single song will occupy two places in the chart with split metrics.

**Comment by LeonNamowitz** [comment]
Tldr: This feature would be much appreciated, as it is almost bound to majorly mess up metrics if you import years long Spotify histories.


Yes some albums have 4, 5, 6, 7 or even more version on Spotify and depending on when you realized this, you may have added all 7 different versions of a single song to different or the same playlist. Resulting in the described behaviour.

Workaround I used is manually going through the Spotify export and consolidation all mismatched IDs of identical songs while simultaneously going through all my playlists and removing duplicate versions while making sure to only keep the one I changed every other to in the json. Then import and never ever add the same old song from a new album to a playlist or even listen to it.

**Comment by LeonNamowitz** [comment]
Example:

##  that won't save us - Against The Current


| listens   | ID (URI)                                 | Album | length |
| --------- | ---------------------------------------- | ----- | ---- |
| **# 180** | spotify:track:4sB0hXFEmjZJR09ilZWsQw     | "that won't save us" | 3:20 |
| **# 68**  | spotify:track:6MlQJCjsxksoruwsgKfLMe | "fever" | 3:20 |
| **# 21**  | spotify:track:28QmTiL06pkEZjSnCl2K7T     | "weapon" | 3:20 |
| **# 13**  | spotify:track:11l31jxa0mUCootwEMAMwR     | "again&again" | 3:20 |


All of these are exactly the same song only on different albums, meaning they show up as 4 individual entries on the metrics because the Spotify song ID is different for some reason.

**Comment by LeviMSmith** [comment]
Been having similar issues, but with the same album, artist, and track name, just different spotify id. I was poking around and found that in those cases the tracks do seem to have the same `external_ids.isrc` field, which according to the spotify docs is the [international standard recording code](https://en.wikipedia.org/wiki/International_Standard_Recording_Code). It seems like a good way to de-duplicate? We could switch stats/counts to use ISRC as the canonical key, and only fall back to Spotify’s track id when ISRC is missing, which in my database is 38 out of ~17k tracks.

@Yooooomi do you think this is a good direction? If so, I’d be happy to take a first pass at a patch.

**Comment by LeonNamowitz** [comment]
Anecdote from [annas-archive.li](https://annas-archive.li/blog/backing-up-spotify.html) for how big of a fucking problem this is..

> Some songs, especially popular songs, have 2, 3, or even 20 different versions. We can quantify this by counting number of songs per [ISRC](https://en.wikipedia.org/wiki/International_Standard_Recording_Code):

<img width="800" height="600" alt="Image" src="https://github.com/user-attachments/assets/c2834e5a-1ddd-4255-accd-c7b32830fcfe" />

**Comment by TheTastyHanuta** [comment]
Hey everyone, 
just wondering whats the status on this? I thought about adding an option to manually merge entries by selecting which ones are the same. This would of course not work automatically for new entries but at least it would work for existing entries as a workaround.
